### ISSUE TITLE
lang: add token account constraint that verifies a token account is "clean"

### ISSUE THREAD

**Issue by paul-schaaf** [solana-foundation/anchor#1341]
`token::clean` -> account must have: no delegate, no close authority

**Comment by paul-schaaf** [comment]
blocked by https://github.com/project-serum/anchor/issues/1222

**Comment by paul-schaaf** [comment]
maybe it would be even better to forbid delegate and close authority by default instead and then have two attributes to enable them if needed.

**Comment by armaniferrante** [comment]
Agreed. Let's force accounts to be clean and require programs to opt in to dirty.
